### PR TITLE
Finger plugin amendments

### PR DESCRIPTION
**Pull request by mgorny-morecode**
#131 finger amendments - Change docker image from `komand/python-2-plugin` to `komand/python-3-37-plugin:3` to use python 3 | Use input and output constants | Changed variables names to more readable | Changed descriptions | Added "f" strings | Removed method test from action

## Proposed Changes
- Change docker image from `komand/python-2-plugin` to `komand/python-3-37-plugin:3` to use python 3
- Use input and output constants
- Changed variables names to more readable
- Changed descriptions
- Added "f" strings
- Removed method test from action

### Description

Describe the proposed changes:

  -

### Testing

Any testing information goes here.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

### Style

Review the [style guide](https://komand.github.io/python/style.html)

- [x] For dependencies, pin [OS package](https://komand.github.io/python/style.html#dockerfile) and [Python package](https://komand.github.io/python/style.html#requirements-txt) versions
- [x] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [x] For size, use the [slim SDK images](https://komand.github.io/python/sdk.html#sdk-versions) when possible: ``komand/python-3-37-slim-plugin`` and ``komand/python-3-37-plugin``
- [x] For error handling, use of [PluginException](https://komand.github.io/python/error_handling.html#plugin-exceptions) and [ConnectionTestException](https://komand.github.io/python/error_handling.html#connection-exceptions)
- [x] For logging, use [self.logger](https://komand.github.io/python/sdk.html#logging)
- [x] For docs, use [changelog style](https://komand.github.io/python/style.html#changelog)
- [x] For docs, validate markdown with ``make validate`` which calls ``mdl`` to lint ``help.md``

### Functional Checklist
- [x] Work fully completed
- [x] Functional
  - [x] Any new actions/triggers include JSON [test files](https://komand.github.io/python/style.html#tests) in the `tests/` directory created with `icon-plugin run -c sample $action > tests/$action.json`
  - [x] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [x] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [x] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -T all --debug --jq` (use PR format at end)
  - [x] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `icon-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `icon-plugin run -R all --debug --jq` (use PR format at end)

### Run

<details>

```
{
  "body": {
    "log": "rapid7/Finger:1.0.2. Step name: finger\nFinger online user list request denied\nFinger online user list denied\nFinger server disable\nSorry, we do not support empty finger queries\nUser not found\nfinger: unknown host: \nfinger: connect: Connection timed out\nfinger: fdopen: \nfinger: tcp/finger: unknown service\nfinger: Out of space.\nfinger: out of space.\nfinger: socket: \nusage: finger [-\nfinger: root: no such user\nIn real life: ???\nfinger: connect: Connection refused\n",
    "meta": {},
    "output": {
      "found": true,
      "fullname": "root",
      "home": "/root",
      "login": "root",
      "loginfrom": "Never logged in",
      "loginstatus": "Never logged in",
      "mailstatus": "No mail",
      "plan": "No plan",
      "shell": "/bin/bash",
      "status": "Success"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/finger:1.0.2 --debug run < tests/finger.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Finger:1.0.2. Step name: finger\nFinger online user list request denied\nFinger online user list denied\nFinger server disable\nSorry, we do not support empty finger queries\nUser not found\nfinger: unknown host: \nfinger: connect: Connection timed out\nfinger: fdopen: \nfinger: tcp/finger: unknown service\nfinger: Out of space.\nfinger: out of space.\nfinger: socket: \nusage: finger [-\nfinger: teststuff: no such user\n",
    "meta": {},
    "output": {
      "found": false,
      "status": "finger: teststuff: no such user"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/finger:1.0.2 --debug run < tests/finger_bad.json
</summary>
</details>

<details>

```
{
  "body": {
    "log": "rapid7/Finger:1.0.2. Step name: finger\nFinger online user list request denied\nFinger online user list denied\nFinger server disable\nSorry, we do not support empty finger queries\nUser not found\nfinger: unknown host: \nfinger: connect: Connection timed out\n",
    "meta": {},
    "output": {
      "found": false,
      "status": "finger: connect: Connection timed out"
    },
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/finger:1.0.2 --debug run < tests/finger_failed_connect_bad.json
</summary>
</details>

### Test

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Finger:1.0.2. Step name: finger\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/finger:1.0.2 --debug test < tests/finger.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Finger:1.0.2. Step name: finger\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/finger:1.0.2 --debug test < tests/finger_bad.json
</summary>
</details>

Autogenerate with:
<details>

```
{
  "body": {
    "log": "rapid7/Finger:1.0.2. Step name: finger\n",
    "meta": {},
    "output": null,
    "status": "ok"
  },
  "type": "action_event",
  "version": "v1"
}

```

<summary>
docker run --rm -i rapid7/finger:1.0.2 --debug test < tests/finger_failed_connect_bad.json
</summary>
</details>

### Make validate
<details>

```
[*] Use ``make menu`` for available targets
[*] Including available Makefiles: ../tools/Makefiles/Helpers.mk ../tools/Makefiles/Colors.mk
--
[*] Running validators
[*] Validating plugin at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator DockerValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 10827.242999999999ms

[*] Validating spec with js-yaml
[SUCCESS] Passes js-yaml spec check


[*] Validating markdown...
[SUCCESS] Passes markdown linting

[*] Validating python files for style...
[SUCCESS] Passes flake8 linting

[*] Validating python files for security vulnerabilities...
[main]  INFO    profile include tests: None
[main]  INFO    profile exclude tests: None
[main]  INFO    cli include tests: None
[main]  INFO    cli exclude tests: None
[main]  INFO    running on Python 3.7.5
Run started:2020-01-02 23:33:55.166667

Test results:
        No issues identified.

Code scanned:
        Total lines of code: 282
        Total lines skipped (#nosec): 0

Run metrics:
        Total issues (by severity):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
        Total issues (by confidence):
                Undefined: 0.0
                Low: 0.0
                Medium: 0.0
                High: 0.0
Files skipped (0):
[SUCCESS] Passes bandit security checks

```

<summary>
make validate
</summary>
</details>

<details>

```

icon-validate . --all
[*] Validating plugin with all validators at .

[*] Running Integration Validators...
[*] Executing validator HelpValidator
[*] Executing validator ChangelogValidator
[*] Executing validator RequiredKeysValidator
[*] Executing validator UseCaseValidator
[*] Executing validator SpecPropertiesValidator
[*] Executing validator SpecVersionValidator
[*] Executing validator FilesValidator
[*] Executing validator TagValidator
[*] Executing validator DescriptionValidator
[*] Executing validator TitleValidator
[*] Executing validator VendorValidator
[*] Executing validator DefaultValueValidator
[*] Executing validator IconValidator
[*] Executing validator RequiredValidator
[*] Executing validator VersionValidator
[*] Executing validator DockerfileParentValidator
[*] Executing validator LoggingValidator
[*] Executing validator ProfanityValidator
[*] Executing validator AcronymValidator
[*] Executing validator JSONValidator
[*] Executing validator OutputValidator
[*] Executing validator RegenerationValidator
[*] Executing validator HelpInputOutputValidator
[*] Executing validator ExceptionValidator
[*] Executing validator CredentialsValidator
[*] Executing validator PasswordValidator
[*] Executing validator PrintValidator
[*] Executing validator ConfidentialValidator
[*] Executing validator DockerValidator
[*] Executing validator URLValidator
[*] Plugin successfully validated!

----
[*] Total time elapsed: 17396.038999999997ms

```

<summary>
icon-validate . --all
</summary>
</details>

### CODE REVIEW

**Comment by lmilby-r7** [comment]
@mgorny-morecode Please check the `Details` for the PluginValidator and fix those findings.
In your local env run `icon-validate . --all` in the plugin root to find what GitHub actions is finding.

File: finger/komand_finger/actions/finger/action.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import komand
-from .schema import FingerInput, FingerOutput
+from .schema import FingerInput, FingerOutput, Input, Output, Component
+
+
```
Comment (gusingh-r7) [inline finger/komand_finger/actions/finger/action.py]: This empty lines can be removed.

Comment (mrinehart-r7) [inline finger/komand_finger/actions/finger/action.py]:
```suggestion
```

File: finger/komand_finger/actions/finger/action.py [hunk 2]
```diff
@@ -1,5 +1,7 @@
 import komand
-from .schema import FingerInput, FingerOutput
+from .schema import FingerInput, FingerOutput, Input, Output, Component
+
```
Comment (mrinehart-r7) [inline finger/komand_finger/actions/finger/action.py]:
```suggestion
```

**Pull request merged**